lib/classifier-reborn/backends/bayes_memory_backend.rb

```diff
@@ -0,0 +1,72 @@
+# Author: Sawood Alam <@ibnesayeed>
+
+module ClassifierReborn
+  class BayesMemoryBackend
+    def initialize
+      @total_words     = 0
+      @total_trainings = 0
+      @category_counts = Hash.new { |h, k| h[k] = {training: 0, word: 0} }
+      @categories      = {}
+    end
+
+    def total_words
+      @total_words
+    end
+
+    def update_total_words(diff)
+      @total_words += diff
+    end
+
+    def total_trainings
```

Comment:
Use `attr_reader` here as well. And they can be combined:

```ruby
attr_reader :total_words, :total_trainings
```

Just a comma-separated list of symbols, where the symbol is the name of the variable and the name of the method.